api/api.proto

```diff
@@ -0,0 +1,73 @@
+syntax = "proto3";
+
+package api;
+
+option go_package = "github.com/ligato/vpp-agent/api";
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+option (gogoproto.messagename_all) = true;
+
+import "google/protobuf/any.proto";
+
+message ModelSpec {
+    string module = 1;
+    string class = 2;
+    string version = 3;
+    string kind = 4;
+}
+
+message Model {
+    google.protobuf.Any any = 1;
```

Comment:
This is slick, but what is the client experience like using Any, particularly in Golang?

Reply:
It's very simple. The `models` package also provides short-hands to make it single call for marshaling/unmarshalling all models: https://github.com/ligato/vpp-agent/blob/0fbdf09e631368e282e2baffd365a212157bc89f/api/models/encoding.go#L29-L68

Reply:
@ondrej-fabry That's super painful actually.

You expect to be able to just use types in go when using things like GRPC... Short-circuiting that by requiring unnatural additional Unmarshal calls is kind of awful.

Among other things, it requires any recipient of an object containing Models to understand its an opaque proto thing.  That's a big usability regression on the current API, which gives you nice usable Go objects at all levels that you can just pass around and use.